app/services/HttpIdentityService.scala

```diff
@@ -72,24 +152,59 @@ class HttpIdentityService(apiUrl: String, apiClientToken: String)(implicit wsCli
     }
   }
 
-  private def headers(request: RequestHeader): List[(String, String)] = List(
+  private def getHeaders(request: RequestHeader): List[(String, String)] = List(
     "X-GU-ID-Client-Access-Token" -> Some(s"Bearer $apiClientToken"),
     "X-GU-ID-FOWARDED-SC-GU-U" -> request.cookies.get("SC_GU_U").map(_.value),
     "Authorization" -> request.headers.get("GU-IdentityToken")
   ).flattenValues
 
+  private def postHeaders(request: RequestHeader): List[(String, String)] = List(
+    "X-GU-ID-Client-Access-Token" -> Some(s"Bearer $apiClientToken"),
+    "Content-Type" -> Some("application/json")
+  ).flattenValues
+
   private def trackingParameters(request: RequestHeader): List[(String, String)] = List(
     "trackingReferer" -> request.headers.get("Referer"),
     "trackingUserAgent" -> request.headers.get("User-Agent"),
     "trackingIpAddress" -> request.clientIp
   ).flattenValues
 
   def getUser(user: IdMinimalUser)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, IdUser] = {
-    get(s"user/${user.id}", headers(req), trackingParameters(req)) { resp =>
+    get(s"user/${user.id}", getHeaders(req), trackingParameters(req)) { resp =>
       (resp.json \ "user").validate[IdUser].asEither.leftMap(_.mkString(","))
     }
   }
 
+  def getUserMaybeGuest(user: IdMinimalUser)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, IdUser] = {
+    get(s"user/${user.id}", getHeaders(req), trackingParameters(req)) { resp =>
+      (resp.json \ "user").validate[IdUser].asEither.leftMap(_.mkString(","))
+    }
+  }
+
+  def getUserIdFromEmail(email: String)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, String] = {
+    get(s"user", getHeaders(req), List("emailAddress" -> email)) { resp =>
+      resp.json.validate[UserResponse].asEither.map(_.user.id).leftMap(_.mkString(","))
+    }
+  }
+
+  def createUserIdFromEmailUser(email: String)(implicit req: RequestHeader, ec: ExecutionContext): EitherT[Future, String, String] = {
+    val body = CreateGuestAccountRequestBody.fromEmail(email)
+    execute(
+      wsClient.url(s"$apiUrl/guest")
+        .withHttpHeaders(postHeaders(req): _*)
+        .withBody(body)
+        .withRequestTimeout(4.second)
```

Comment:
Why 4 seconds?

Reply:
good catch, this was left over from development. I'll revert to 1 second, as is the current behaviour in the `get` method

Reply:
👍